app/controllers/api/v2/fact_values_controller.rb

```diff
@@ -15,7 +15,8 @@ def index
           no_timestamp_facts.
           search_for(*search_options).paginate(paginate_options).
           preload(:fact_name, :host)
-        @fact_values = FactValue.build_facts_hash(values.all)
+        host_id = params[:host_id] || nil
```

Comment:
You don't need to specify this here, if there is no such param then `params[:host_id]` will already return nil.